### PR TITLE
Create section for developers in Cosmology docs

### LINKED ISSUE #12099: Create section for developers in Cosmology docs
Start by moving developer notes in the docs to a dedicated page.

### PR DESCRIPTION
**Pull request by nstarman**
Developer notes in the code and docs are moved to a dedicated page.
Followup will be adding a section explaining how a 3rd party package can plug into astropy's cosmology IO.

Fixes #12099 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
A good idea to separate out developer information. I've long wanted to have some standard way to do it (see #8930), and thought that perhaps the module docstrings were the right place (we barely use them, after all). Here, that might make particular sense for the integrals - may have the advantage that it is less likely to get out of date.

But your choice is definitely fine too, and a real improvement over what's here, so approving as is!

**Comment by nstarman** [comment]
I like that idea. I'd forgotten about https://github.com/astropy/astropy/issues/8930, but it's well worth doing. I made a [card](https://github.com/astropy/astropy/projects/7#card-67361545) to do this as a followup, because I had an issue with section title levels when I just tried.

File: docs/changes/cosmology/12100.other.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Developer notes in the code and docs are moved to a dedicated page.
```
Comment (pllim) [inline docs/changes/cosmology/12100.other.rst:1]: I think this fragment name violates the towncrier rules. See #11945

Comment (nstarman) [reply docs/changes/cosmology/12100.other.rst:1]: Ah. Thanks. I'll push a fix.
What rule specifically?

**Pull request merged**